fix: Rust now correctly highlights doc comments as @comment.block.documentation

CalebLarsen opened the pull request:
Closes #13658.

This change makes `/// Example` and `//! Example` all match as `@comment.block.documentation`.
Previously, these cases would all match as `@comment`.

the-mikedavis commented on runtime/queries/rust/highlights.scm:
```diff
@@ -106,12 +106,17 @@
   (line_comment)
   (block_comment)
 ] @comment
+(line_comment
+  (outer_doc_comment_marker "/" @comment.block.documentation)
+  (doc_comment)) @comment.block.documentation
+(line_comment
```
For these we should tag as `comment.line.documentation` - Rust also has block doc comments ([reference](https://doc.rust-lang.org/reference/comments.html)) as

```rust
/**  - Outer block doc (exactly) 2 asterisks */
```

which we could tag as well

CalebLarsen commented:
Updated to support both line and block doc comments. I also added a commit that went through themes that defined "comment.block.documentation" and added "comment.line.documentation". I tried to make the sensible call and follow what the theme was trying to achieve.
